### PR TITLE
fix(react-router): redirect with relative path

### PR DESCRIPTION
**Pull request by schiller-manuel**
tests needed

### CODE REVIEW

File: packages/react-router/tests/link.test.tsx [hunk 1]
```diff
@@ -2527,6 +2527,204 @@ describe('Link', () => {
...
+
+    const postRoute = createRoute({
+      getParentRoute: () => postsRoute,
+      path: '$postId',
+      component: PostComponent,
+      validateSearch: () => ({ postPage: 0 }),
+      loader: () => {
+        throw redirect({
+          from: postRoute.fullPath,
+          to: '/../../login',
```
Comment (schiller-manuel) [inline packages/react-router/tests/link.test.tsx:2587]: why is there a leading slash?

File: packages/react-router/tests/link.test.tsx [hunk 2]
```diff
@@ -2527,6 +2527,204 @@ describe('Link', () => {
...
+
+    const postRoute = createRoute({
+      getParentRoute: () => postsRoute,
+      path: '$postId',
+      component: PostComponent,
+      validateSearch: () => ({ postPage: 0 }),
+      beforeLoad: () => {
+        throw redirect({
+          from: postRoute.fullPath,
+          to: '/../../login',
```
Comment (schiller-manuel) [inline packages/react-router/tests/link.test.tsx:2686]: same here

File: packages/react-router/tests/link.test.tsx (hunk 1 above)
Comment (chorobin) [reply packages/react-router/tests/link.test.tsx:2587]: True! weird it works with it

**Review by SeanCassiere — COMMENTED** [review]
I think adding into a relative navigation to another route in the same route level wouldn't hurt similar to the example @schiller-manuel posted on Discord.

Something along the lines of:

```
from: /wherever/
to: ./somewhere
```

**Pull request merged**